documentation/docs/proptest/assumptions.md

```diff
@@ -0,0 +1,97 @@
+---
+id: assumptions
+title: Assumptions
+slug: property-test-assumptions.html
+sidebar_label: Assumptions
+---
+
+
+To restrict the set the generated values from a generator, we can use filter to return a constrained arb.
+
+```kotlin
+val evens = Arb.int().filter { it.value % 2 == 0 }
+```
+
+However if we need to restrict inputs based on the relationship between values, then filtering on a single arb won't
+work. Consider an example where we want to test that two non-equal strings have a non-zero Levenshtein distance.
+
+```kotlin
+checkAll<String, String> { a, b ->
+  levenshtein(a, b) shouldBeGreaterThan 0
+}
+```
+
+This will periodically fail - whenever two equal strings are generated. One approach would be to just wrap the tests in
+an
+if/else block and avoid those undesired inputs.
+
+```kotlin
+checkAll<String, String> { a, b ->
+  if (a != b)
+    levenshtein(a, b) shouldBeGreaterThan 0
+}
+```
+
+But in more complicated scenarios we could easily introduce a bug and filter _all_ our inputs.
+
+Kotest provides a feature called _assumptions_ that will filter out unwanted combinations, while tracking that we
+are not filtering too many.
+
+An assumption accepts a boolean value passed to the `withAssumptions` function, that if true, will allow the property
+test to continue, but if false, that particular iteration is skipped. For example, the previous example will now pass:
+
+```kotlin
+checkAll<String, String> { a, b ->
+  withAssumptions(a != b) {
+    levenshtein(a, b) shouldBeGreaterThan 0
+  }
+}
+```
+
+### Assertions
+
+Kotest expands on basic boolean assumptions by allowing you to specify any assertion in the assumption predicate, in
+addition to supporting an arbitrary number of assumptions.
+
+For example, building on the previous example:
+
+```kotlin
+checkAll(Arb.string(3..4, Codepoint.az()), Arb.string(3..4, Codepoint.az())) { a, b ->
+  withAssumptions(
+     { a shouldNotBe b },
+     { a shouldHaveLength (b.length) },
```

Comment:
I supposed these assertions shouldn't be counted in the AssertionsCounter, mainly for the purposes of not falsely counting assertions if no test is actually made